sbpy/data/phys.py

```diff
@@ -153,6 +164,136 @@ def from_sbdb(cls, targetids, references=False, notes=False):
+
+        """
+        try:
+            from scipy import interpolate
+        except ImportError:
+            raise ImportError('Optional package scipy is needed for this \
+                               function. Please install scipy.')
+
+        if isinstance(mol_tag, str):
+            query = JPLSpec.query_lines_async(min_frequency=(transition_freq - (1 * u.GHz)),
+                                              max_frequency=(transition_freq + (1 * u.GHz)),
+                                              molecule=mol_tag,
+                                              parse_name_locally=True,
+                                              get_query_payload=True)
+
+            res = dict(query)
+            # python request payloads aren't stable (could be dictionary or list)
+            # depending on the version, so make sure to check back from time to time
+            if len(res['Mol']) > 1:
+                raise JPLSpecQueryFailed(("Ambiguious choice for molecule,\
+                                         more than one molecule was found for \
+                                         the given mol_tag. Please refine \
+                                         your search to one of the following tags\
+                                         {} by using JPLSpec.get_species_table()\
+                                         (as shown in JPLSpec documentation)\
+                                         to parse their names and choose your \
+                                         molecule of interest, or refine your\
+                                         regex to be more specific.\
+                                         ").format(res['Mol']))
+
+        query = JPLSpec.query_lines(min_frequency=(transition_freq - (1 * u.GHz)),
+                                    max_frequency=(transition_freq + (1 * u.GHz)),
+                                    molecule=mol_tag)
+
+        freq_list = query['FREQ']
+
+        if freq_list[0] == 'Zero lines we':
+            raise JPLSpecQueryFailed("Zero lines were found by JPLSpec in \
+                                       a +/- 1 GHz range from your provided \
+                                       transition frequency.")
+
+        t_freq = min(list(freq_list.quantity),
+                     key=lambda x: abs(x-transition_freq))
+
+        data = query[query['FREQ'] == t_freq.value]
+
+        df = int(data['DR'].data)
+
+        lgint = float(data['LGINT'].data)
+
+        lgint = 10**lgint * u.nm * u.nm * u.MHz
+
+        elo = float(data['ELO'].data) / u.cm
+
+        gu = float(data['GUP'].data)
+
+        cat = JPLSpec.get_species_table()
+
+        mol = cat[cat['TAG'] == mol_tag]
+
+        temp_list = cat.meta['Temperature (K)'] * u.K
+
+        part = list(mol['QLOG1', 'QLOG2', 'QLOG3', 'QLOG4', 'QLOG5', 'QLOG6',
+                        'QLOG7'][0])
+
+        temp = temp_estimate
+
+        f = interpolate.interp1d(temp_list, part, 'linear')
```

Comment:
Linear interpolation can also be done with `numpy.interp`.  If you use numpy instead, scipy wouldn't be needed.

Reply:
You're correct, I only recently just learned numpy had a function for linear interpolation, and I can definitely apply it here.